src/game/Spells/Spell.cpp

```diff
@@ -1299,6 +1299,7 @@ uint64 Spell::CalculateDelayMomentForDst() const
 void Spell::RecalculateDelayMomentForDest()
 {
     m_delayMoment = CalculateDelayMomentForDst();
+    m_destTargetInfo.timeDelay = 0;
```

Comment:
Based on my code analysis, m_destTargetInfo.timeDelay = m_delayMoment;

It designates the time dest effects should go off, GetDelayStart does not change.

Reply:
timeDelay is an independent timer that ticks down once the Event that's set a line below fires. it goes through a while timeDelay >= diff loop
so when that delay is greater than 0 you will not only have the delay that comes from the event, but then also the additional delay that comes from this variable on top